saw-remote-api/python/saw/__init__.py

```diff
@@ -421,6 +432,21 @@ def llvm_verify(module: LLVMModule,
 
     return result
 
+def prove(goal: cryptoltypes.CryptolJSON,
+          proof_script: proofscript.ProofScript) -> ProofResult:
+    conn = __get_designated_connection()
+    res = conn.prove(cryptoltypes.to_cryptol(goal),
+                     proof_script.to_json()).result()
+    pr = ProofResult()
+    if res['status'] == 'valid':
+        pr.valid = True
+    elif res['status'] == 'invalid':
+        pr.valid = False
+    else:
+        raise ValueError("Unknown proof result " + str(res))
+    if 'counterexample' in res:
+        pr.counterexample = res['counterexample']
```

Comment:
I was just looking over this again -- would it be better to do more processing here for the returned counter examples? E.g., so that it's a Python list of values instead of a JSON-esque dictionary? (Basically I'm biased to hide the raw JSON as much as possible from users, and this seems like it might be once place it could leak through.)

I was just playing around with adding `:check` to the Cryptol python interface, and there it seemed convenient to convert the returned counterexamples into the argument list in terms of python values, i.e.:
```
elif res['result'] == 'fail':
    return CheckReport(
            success=False,
            args=[from_cryptol_arg(arg['expr']) for arg in res['arguments']], # <-- here
            error_msg = None,
            tests_run=res['tests run'],
            tests_possible=res['tests possible'])
```
(Well it has _seemed_ nice so far anyway -- there's obviously still not a lot of examples of those returned values being of use per se.)

Reply:
Yeah, I agree with this, and just hadn't concluded what the best high-level data structure would be best. I'll give it some more thought.